src/zenml/orchestrators/step_launcher.py

```diff
@@ -296,10 +374,11 @@ def _bypass() -> None:
                         artifacts=step_run.outputs,
                         tags=pipeline_run.config.tags,
                     )
-
+        except RunStoppedException:
+            logger.info(f"Pipeline run `{pipeline_run.name}` stopped.")
+            raise
         except:  # noqa: E722
             logger.error(f"Pipeline run `{pipeline_run.name}` failed.")
-            publish_utils.publish_failed_pipeline_run(pipeline_run.id)
```

Comment:
So if the step failed for any other reason, like an error in the custom user code, you would previously mark the pipeline as failed, but not anymore. Why ?

And it should have been the step being marked as failed, not the entire pipeline run, correct ?

Reply:
1. I don't think that we need this here anymore. Every time a step's status gets updated, the server actually updates the status of the pipeline run accordingly anyway.
2. Since the exception is still raised, this step will still be marked as `Failed`.

Reply:
I think we never needed this, but it was added as an additional safety mechanism if for some reason the step status update failed.